File: music-application/src/main/java/com/music/sale/application/user/dto/input/CreateSocialUserInput.java
package com.music.sale.application.user.dto.input;

import com.music.sale.domain.user.enums.SocialProvider;

public record CreateSocialUserInput(
    SocialProvider provider,
    String providerId,
    String email,
    String nickname,
    String profileImageUrl) {

  public static CreateSocialUserInput of(
      SocialProvider provider,
      String providerId,
      String email,
      String nickname,
      String profileImageUrl) {
    return new CreateSocialUserInput(provider, providerId, email, nickname, profileImageUrl);
  }
}

File: music-application/src/main/kotlin/com/music/sale/application/user/dto/input/CreateSocialUserInput.kt
package com.music.sale.application.user.dto.input

import com.music.sale.domain.user.enums.SocialProvider

data class CreateSocialUserInput(
    val provider: SocialProvider,
    val providerId: String,
    val email: String,
    val nickname: String,
    val profileImageUrl: String,
) {
    companion object {
        @JvmStatic
        fun of(
            provider: SocialProvider,
            providerId: String,
            email: String,
            nickname: String,
            profileImageUrl: String,
        ): CreateSocialUserInput = CreateSocialUserInput(provider, providerId, email, nickname, profileImageUrl)
    }
}
